pkg/netpol/eval/check_eval.go

```diff
@@ -0,0 +1,257 @@
+		return anpRes, nil
+	}
+	// else pass == true : means that:
+	// - the admin-network-policies did not capture the connection (if the ANPs rules did not capture the connection explicitly,
+	// then it is not captured.)
+	// - or that the rules captured the connection with action: pass; so it will be determined with netpols/ banp.
+	netpolRes, captured, err := pe.allowedXgressConnectionByNetpols(src, dst, isIngress, protocol, port)
+	if err != nil {
+		return false, err
+	}
+	// if the src/dst was captured by the relevant xgress policies, then the connection is
+	// definitely allowed or denied by the policy rules (either explicitly or implicitly)
+	if captured {
+		pe.cache.addConnectionResult(src, dst, protocol, port, netpolRes)
+		return netpolRes, nil
+	}
+	// else !captured : means that the xgress connection will be determined by the baseline-admin-network-policy,
+	// or the system-default for connection that was not captured by any policy which is allowed.
+	defaultRes, err := pe.allowedXgressByBaselineAdminNetpolOrByDefault(src, dst, isIngress, protocol, port)
+	if err != nil {
+		return false, err
+	}
+	pe.cache.addConnectionResult(src, dst, protocol, port, defaultRes)
+	return defaultRes, nil
+}
+
+// allowedXgressConnectionByAdminNetpols returns if the given input connection is allowed on the given ingress/egress direction
+// by k8s admin-network-policies
+func (pe *PolicyEngine) allowedXgressConnectionByAdminNetpols(src, dst k8s.Peer, isIngress bool, protocol, port string) (res, pass bool,
+	err error) {
+	// iterate sorted by priority admin netpols
+	for _, anp := range pe.sortedAdminNetpols {
+		if isIngress {
+			selectsDst, err := anp.Selects(dst, true)
+			if err != nil {
+				return false, false, err
+			}
+			if selectsDst {
+				res, captured, err := anp.CheckIngressConnAllowed(src, dst, protocol, port)
+				if err != nil {
+					return false, false, err
+				}
+				if !captured {
+					continue // continue to next ANP
+				}
+				return analyzeANPCapturedRes(res)
+			}
+		} else { // egress
+			selectsSrc, err := anp.Selects(src, false)
+			if err != nil {
+				return false, false, err
+			}
+			if selectsSrc {
+				res, captured, err := anp.CheckEgressConnAllowed(dst, protocol, port)
+				if err != nil {
+					return false, false, err
+				}
+				if !captured {
+					continue // continue to next ANP
+				}
+				return analyzeANPCapturedRes(res)
+			}
+		}
+	}
+	// getting here means the connection was not captured by any ANP - pass to netpols
+	return false, true, nil
+}
+
+// analyzeANPCapturedRes when an admin-network-policy captures a connection , its result may be Allow (final- allowed conn),
+// or Deny (final - denied conn) or Pass (to be determined by netpol/ banp)
+func analyzeANPCapturedRes(anpRes int) (allowedOrDenied, pass bool, err error) {
+	switch anpRes {
+	case k8s.Pass:
+		return false, true, nil
+	case k8s.Allow:
+		return true, false, nil
+	case k8s.Deny:
+		return false, false, nil
```

Comment:
why is the first return arg named `allowedOrDenied`, but for the case of `Deny` it returns `false`?

Reply:
the name of the variable is  "allowedOrDenied" i.e. the connection is either allowed or denied
in case it is allowed then the result for the given connection is true;
but if it is deny then the result is false

Reply:
ok, please explain in function documentation

Reply:
done